vortex-array/src/array/primitive/compute/subtract_scalar.rs

```diff
@@ -60,31 +58,12 @@ fn subtract_scalar_integer<
         return Ok(subtract_from.clone());
     }
 
-    if let Some(min) = subtract_from.statistics().compute_as_cast::<T>(Stat::Min) {
-        if let (_, true) = min.overflowing_sub(&to_subtract) {
-            vortex_bail!(
-                "Integer subtraction over/underflow: {}, {}",
-                min,
-                to_subtract
-            )
-        }
-    }
-    if let Some(max) = subtract_from.statistics().compute_as_cast::<T>(Stat::Max) {
-        if let (_, true) = max.overflowing_sub(&to_subtract) {
-            vortex_bail!(
-                "Integer subtraction over/underflow: {}, {}",
-                max,
-                to_subtract
-            )
-        }
-    }
-
     let contains_nulls = !subtract_from.logical_validity().all_valid();
     let subtraction_result = if contains_nulls {
         let sub_vec = subtract_from
             .maybe_null_slice()
             .iter()
-            .map(|&v: &T| v.saturating_sub(&to_subtract))
+            .map(|&v: &T| v.wrapping_sub(&to_subtract))
```

Comment:
why are we changing the behavior of this, and removing the assertions?

Reply:
The assertions are expensive since they require computing stats.
We could either just use checked subtraction and allow this to panic, or we pick between saturating and wrapping. 

Tbh, I've never seen unchecked subtraction do anything other than wrap, so it seemed like the better choice. e.g. https://arrow.apache.org/docs/python/generated/pyarrow.compute.subtract.html